### ISSUE TITLE
Set rate limits on the processing log

### ISSUE THREAD

**Issue by apurvam** [confluentinc/ksql#6775]
**Is your feature request related to a problem? Please describe.**
Curretnly the processing log has unbound throughput on kafka, and can also cause exccessive disk usage since the default retention is one day. 

**Describe the solution you'd like**
We should set limits on the processing log to protect against high error rate / high throughput queries from causing resource exhaution issues on kafka.

**Comment by agavra** [comment]
We should consider bytes retention instead of time retention.

**Comment by agavra** [comment]
@spena does the rate limit on the logger prevent processing of messages? that might be acceptable, but it's something to note. Also this seems like a promising approach! If so, we should document it in our public docs so that other users can follow along :)

**Comment by spena** [comment]
Yes, it prevents processing all messages. The limiter sleeps for a few seconds before sending the message to kafka.

**Comment by agavra** [comment]
Let's make sure that if we're throttling the user this is something that shows up at the top level - maybe something like query status throttled?

**Comment by rodesai** [comment]
@vcrfxia why did we unassign this and label core-infra?

**Comment by vcrfxia** [comment]
> @vcrfxia why did we unassign this and label core-infra?

Responded offline -- KSE didn't have capacity as of our last planning meeting. Latest update on the ticket (at the time) was that Sergio merged cc-docker-ksql changes for dropping messages if the processing log logs above a certain rate but hadn’t made spec changes to set a limit. It's also worth exploring the other idea (on this ticket) of configuring bytes-based retention on the processing log topic, rather than relying on throttling.
